src/python/gudhi/representations/vector_methods.py

```diff
@@ -807,20 +807,28 @@ def fit(self, X, y=None, sample_weight=None):
             self
         """
         if not hasattr(self.quantiser, 'fit'):
-            raise TypeError("quantiser %s has no `fit` attribute." % (self.quantiser))
-
-        # In fitting we remove infinite death time points so that every center is finite
-        X = [dgm[~np.isinf(dgm).any(axis=1), :] for dgm in X]
+            raise TypeError(f"quantiser {self.quantiser} has no `fit` attribute.")
 
         if sample_weight is None:
             sample_weight = [self.get_weighting_method()(measure) for measure in X]
 
         measures_concat = np.concatenate(X)
         weights_concat = np.concatenate(sample_weight)
-
-        self.quantiser.fit(X=measures_concat, sample_weight=weights_concat)
-
+        # In fitting we remove infinite birth/death time points so that every center is finite
+        filtered_measures_concat = measures_concat[~np.isinf(measures_concat).any(axis=1), :] if len(measures_concat) else measures_concat
+        filtered_weights_concat = weights_concat[~np.isinf(measures_concat).any(axis=1)] if len(measures_concat) else weights_concat
+
+        n_clusters = self.quantiser.n_clusters
+        n_points = len(filtered_measures_concat)
+        if n_points < n_clusters:
+            # If not enough points to fit (including 0), let's arbitrarily put centers in [0, 1)^2
+            print(f"[Atol] had {n_points} points to fit {n_clusters} clusters, adding random points in [0, 1)^2.")
```

Comment:
Wait, is Atol restricted to 2D? I thought it wasn't restricted to diagrams and could work in any dimension. Though if we fit on no points, I don't know how to guess a dimension for the random centers.

Reply:
It is not, this is a hard one!

Reply:
There is still the possibility of giving up (i.e. raise an exception if there are no points at all).

Reply:
I have now favored and implemented this approach!